File: app/src/main/java/com/pashkobohdan/ttsreader/mvp/bookList/view/BookListView.java
package com.pashkobohdan.ttsreader.mvp.bookList.view;

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.pashkobohdan.ttsreader.model.dto.book.BookDTO;
import com.pashkobohdan.ttsreader.mvp.bookList.BookListPresenter;
import com.pashkobohdan.ttsreader.mvp.common.AbstractScreenView;

import java.util.List;

/**
 * Created by bohdan on 07.08.17.
 */
@StateStrategyType(SkipStrategy.class)
public interface BookListView extends AbstractScreenView<BookListPresenter> {

    @StateStrategyType(AddToEndSingleStrategy.class)
    void showBookList(List<BookDTO> bookDTOList);

    void addBook(BookDTO bookDTO);

    void deleteBook(int deletedBookPosition);

    void showDataExecutionError();

    void showRefresh();

    void hideRefresh();
}

File: app/src/main/java/com/pashkobohdan/ttsreader/mvp/bookList/view/BookListView.kt
package com.pashkobohdan.ttsreader.mvp.bookList.view

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.pashkobohdan.ttsreader.model.dto.book.BookDTO
import com.pashkobohdan.ttsreader.mvp.bookList.BookListPresenter
import com.pashkobohdan.ttsreader.mvp.common.AbstractScreenView
import java.io.File

@StateStrategyType(SkipStrategy::class)
interface BookListView : AbstractScreenView<BookListPresenter> {

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun showBookList(bookDTOList: List<BookDTO>)

    fun showEditBook(bookDTO: BookDTO)

    fun showGetBookListError()

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun showProgress()

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun hideProgress()

    fun bookSaveSuccess()

    fun bookSaveError(bookDTO: BookDTO)

    fun bookRemoveSuccess()

    fun bookRemoveError(bookDTO: BookDTO)

    fun showFileOpenerChooser()

    fun showBookOpenDialog(file: File)

    fun showBookIsAlreadyExistError()
}
